pytorch/training/docker/2.1/py3/cu121/Dockerfile.gpu

```diff
@@ -366,6 +368,11 @@ RUN /opt/conda/bin/mamba install -y python=${PYTHON_SHORT_VERSION} pytorch=${PYT
   -c conda-forge \
   && /opt/conda/bin/mamba clean -afy
 
+
+# Install SageMaker Profiler Binary
+RUN apt-get update && apt-get install -y libunwind-dev && apt-get clean \
+ && pip install --no-cache-dir -U ${SMP_URL}
```

Comment:
You will need to declare `SMP_URL` as another `ARG` at the top of this section of the dockerfile (refer to lines 351-358 in this dockerfile), after having defined a value for it above.

Please refer to how `PYTHON` is passed as an `ARG` in this dockerfile.